src/mltcontroller.cpp

```diff
@@ -557,6 +559,8 @@ QString Controller::XML(Service *service, bool withProfile, bool withMetadata)
     c.start();
```

Comment:
The connect function immediately above this comment is where the consumer on the service is being replaced with the XML consumer. Before this call, the consumer is the multitrack (for a track). After this function closes, the XML consumer is deleted. But the track service still holds a reference. Not sure about how other services use the consumer internally. It must not be a problem for clips since we use this function regularly on clips.